plugins/json-multisheet-extractor/src/parser.ts

```diff
@@ -0,0 +1,42 @@
+import { WorkbookCapture, parseSheet } from '@flatfile/util-extractor'
+
+
+export function parseBuffer(buffer: Buffer): WorkbookCapture {
+  try {
+    const fileContents = buffer.toString('utf8')
+
+    if (!fileContents) {
+      console.log('Invalid file contents')
+      return {} as WorkbookCapture
+    }
+
+    const parsedData = JSON.parse(fileContents)
+    if (typeof parsedData !== 'object' || parsedData === null) {
+      console.error("Invalid input: data must be an object.");
+      return {};
+    }
+
+    const results = {};
+
+    Object.keys(parsedData).forEach(sheet => {
+      const value = parsedData[sheet];
+
+      // Check if key is a string and value is an array
+      if (typeof sheet === 'string' && Array.isArray(value)) {
+          try {
+              const result = parseSheet(value);
+              results[sheet] = result; // Store result in the results object
+          } catch (error) {
+              console.error(`Error processing sheet "${sheet}":`, error);
+          }
+      } else {
+          console.warn(`Skipping key "${sheet}" - not a string/array pair.`);
+      }
+    });
```

Comment:
AI initiated thread on error handling https://github.com/FlatFilers/flatfile-plugins/pull/693/files#r1826003554

1. Array type validation is delegated to the parseSheet function
2. Failing fast is not necessarily the best approach. Requires Flatfile product input. Personally, I'd rather pull out as much data as possible and allow for a partial success than a complete failure. We could likely use more nuanced response types to address this (e.g. sheet error is returned as part of the SheetCapture object for surfacing to the user, otherwise they might not know what part of a potentially large JSON file is broken)
3. Will set the type for results to WorkbookCapture as opposed to the type another level below (the AI eventually suggests `Record<string, SheetCapture>`, which is effectively WorkbookCapture)